bites_test.py

```diff
@@ -635,5 +652,229 @@ def test_play_full_game_calls_take_all_turns_first_and_then_print_scores(
+    self.assertEqual(print_mock.call_args_list[7], mock.call("cheese"))
+    self.assertEqual(print_mock.call_args_list[8], mock.call("bread"))
+    print_patcher.stop()
+
+  def test_check_render_game_prints_players_and_trail(self):
+    # test 89
+    class FakePlayer():
+      def __init__(self, name):
+        self.name = name
+        self.hand = {}
+
+    fake_mario = FakePlayer("mario")
+    fake_luigi = FakePlayer("luigi")
+
+    ants = []
+    tokens_for_trail = {}
+    players = [fake_mario, fake_luigi]
+    bites_game = Bites(ants, tokens_for_trail, players)
+    bites_game.trail = ["pepper", "apple", "grapes", "cheese", "bread"]
+    bites_game.ant_positions = {"random key": None}
+
+    print_patcher = mock.patch('builtins.print')
+    print_mock = print_patcher.start()
+    bites_game.render_game()
+    self.assertGreaterEqual(print_mock.call_count, 7)
+    self.assertEqual(print_mock.call_args_list[1], mock.call("mario: {}"))
+    self.assertEqual(print_mock.call_args_list[2], mock.call("luigi: {}"))
+    self.assertEqual(print_mock.call_args_list[6], mock.call("pepper"))
+    self.assertEqual(print_mock.call_args_list[7], mock.call("apple"))
+    self.assertEqual(print_mock.call_args_list[8], mock.call("grapes"))
+    self.assertEqual(print_mock.call_args_list[9], mock.call("cheese"))
+    self.assertEqual(print_mock.call_args_list[10], mock.call("bread"))
+    print_patcher.stop()
+
+  def test_render_game_prints_ants_positioned_before_the_trail(self):
+    # test 90
+    ants = []
+    tokens_for_trail = {}
+    players = []
+    bites_game = Bites(ants, tokens_for_trail, players)
+    bites_game.ant_positions = {"red": None}
+
+    print_patcher = mock.patch('builtins.print')
+    print_mock = print_patcher.start()
+    bites_game.render_game()
+    self.assertGreaterEqual(print_mock.call_count, 3)
+    self.assertEqual(print_mock.call_args_list[0], mock.call("\nPlayer names and hands:"))
+    self.assertEqual(print_mock.call_args_list[1], mock.call("\nAnts at the beginning of the trail:"))
+    self.assertEqual(print_mock.call_args_list[2], mock.call("red"))
+    print_patcher.stop()
+
+  def test_render_game_shows_ants_positioned_on_trail_in_correct_place(self):
+    # test 91
+    ants = []
+    tokens_for_trail = {"apple" :1}
+    players = []
+    bites_game = Bites(ants, tokens_for_trail, players)
+    bites_game.ant_positions = {"red": 0}
+
+    print_patcher = mock.patch('builtins.print')
+    print_mock = print_patcher.start()
+    bites_game.render_game()
+    self.assertGreaterEqual(print_mock.call_count, 3)
+    self.assertEqual(print_mock.call_args_list[0], mock.call("\nPlayer names and hands:"))
+    self.assertEqual(print_mock.call_args_list[1], mock.call("\nTrail and ant positions:"))
+    self.assertEqual(print_mock.call_args_list[2], mock.call("apple red"))
+    print_patcher.stop()
+
+  def test_check_render_game_shows_player_names_and_hands__ants_waiting_to_start__and_trail_with_ants(
```

Comment:
Are the double underscores used on purpose?

Reply:
Yes. I wrote it like this because the name is long and is checking for three different things, it's in lieu of being able to include proper punctuation.